barretenberg/cpp/src/barretenberg/translator_vm/translator_prover.cpp

```diff
@@ -2,66 +2,20 @@
 #include "barretenberg/commitment_schemes/claim.hpp"
 #include "barretenberg/commitment_schemes/commitment_key.hpp"
 #include "barretenberg/commitment_schemes/shplonk/shplemini.hpp"
-#include "barretenberg/honk/proof_system/permutation_library.hpp"
 #include "barretenberg/plonk_honk_shared/library/grand_product_library.hpp"
 #include "barretenberg/sumcheck/sumcheck.hpp"
 
 namespace bb {
 
-TranslatorProver::TranslatorProver(CircuitBuilder& circuit_builder,
-                                   const std::shared_ptr<Transcript>& transcript,
-                                   std::shared_ptr<CommitmentKey> commitment_key)
-    : dyadic_circuit_size(Flavor::compute_dyadic_circuit_size(circuit_builder))
-    , mini_circuit_dyadic_size(Flavor::compute_mini_circuit_dyadic_size(circuit_builder))
-    , transcript(transcript)
-    , key(std::make_shared<ProvingKey>(circuit_builder))
+TranslatorProver::TranslatorProver(const std::shared_ptr<TranslatorProvingKey>& key,
+                                   const std::shared_ptr<Transcript>& transcript)
+    : transcript(transcript)
+    , key(key)
 {
     PROFILE_THIS();
-
-    key->commitment_key = commitment_key ? commitment_key : std::make_shared<CommitmentKey>(key->circuit_size);
-    compute_witness(circuit_builder);
-}
-
-/**
- * @brief Compute witness polynomials
- *
- */
-void TranslatorProver::compute_witness(CircuitBuilder& circuit_builder)
-{
-    if (computed_witness) {
-        return;
+    if (key->proving_key->commitment_key == nullptr) {
```

Comment:
Ternaries and`nullptr` as true value `false` are pretty standard  but I guess this is fine too.